libvirt/resource_libvirt_domain.go

```diff
@@ -214,6 +214,25 @@ func resourceLibvirtDomain() *schema.Resource {
 								Type: schema.TypeString,
 							},
 						},
+
+						"vlan": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Required: false,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"tag": {
```

Comment:
Yes, thats something that came to my mind too. As each tag might have a different nativeMode option i think it would be better to make it a map. For my use case it didnt make any difference as i dont need to set an nativeMode.